sdk/core/azure-core/inc/credentials/credentials.hpp

```diff
@@ -36,10 +39,10 @@ namespace Azure { namespace Core { namespace Credentials {
   private:
     static std::string const g_aadGlobalAuthority;
 
-    std::string const m_tenantId;
-    std::string const m_clientId;
-    std::string const m_clientSecret;
-    std::string const m_authority;
+    std::string m_tenantId;
```

Comment:
Why do we want to modify these?  I'd expect you initialize in the constructor and never change them after.  If you wanted a credential with a different value then you make a new credential

Reply:
https://azure.github.io/azure-sdk/cpp_design.html#const-and-reference-members + Billy doesn't recommend these due to unexpected UB described in CWG1116 and CWG1776.

Reply:
I've added an agenda item to Tuesday's meeting for const on non-union members.  We can add this back after that discussion.